src/controllers/userController.ts

```diff
@@ -62,4 +63,75 @@ const login = async (
   res.status(StatusCodes.OK).json({ user: { name: user.name }, token });
 };
 
-export { register, login };
+const getUserProfile = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+
+    const { email } = req.body;
+
+    if (!email) {
```

Comment:
i am not shure that we need email here, we could rewrite like this: const email = req.query.email as string;

    if (email) {
        userProfile = await User.findOne({ email: email });
    }else{ 
        userProfile = await User.findOne({ _id: user.userId });
    }

Reply:
Fixed